.github/workflows/gt4py-precommit.yml

```diff
@@ -0,0 +1,24 @@
+name: Formatting & compliance (pre-commit)
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v1
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip pre-commit
+        pre-commit install-hooks
```

Comment:
I don't think `install-hooks` is necessary since you don't commit, but run on all files.

Reply:
Maybe I'm wrong here, but I think that `install-hooks` is actually creating the venvs for running the hooks, not installing the git hooks, so it makes sense to run the command in advance

Reply:
@egparedes is right. The same would also happen on `pre-commit run --all-files` implicitly. However this way the envs can potentially cached? I don't know how this works on gh actions.

Reply:
Ok, I don't know about caches, but it can make sense anyway to have a separate step, so if something fails, you see if it's really in the actual check or something else.